ast.go

```diff
@@ -1223,6 +1262,37 @@ func (node *FuncExpr) walkSubtree(visit Visit) error {
 	return Walk(visit, node.Name, node.Args, node.Filter)
 }
 
+// CustomFuncExpr represents a function call.
+type CustomFuncExpr struct {
```

Comment:
I created this new node to distinguish it from a regular function call. It does not have the `Distinct` and `Filter` attributes and it has a special `ResolvedString` attribute that is used to override the default `String` behavior. This field is set by `Resolve` logic ([write](https://github.com/tablelandnetwork/go-sqlparser/pull/42/files#diff-0945956cd950d047e7fa866dcf34fee7d29cbce277d4e5e5cd621f2213f3be6cR2122) and [read](https://github.com/tablelandnetwork/go-sqlparser/pull/42/files#diff-0945956cd950d047e7fa866dcf34fee7d29cbce277d4e5e5cd621f2213f3be6cR2078)) if all rules apply